python_modules/dagster/dagster/_core/definitions/multi_dimensional_partitions.py

```diff
@@ -334,6 +338,124 @@ def get_partition_keys(
             current_time or get_current_datetime(), dynamic_partitions_store
         )
 
+    def get_partition_key_connection(
+        self,
+        context: PartitionLoadingContext,
+        limit: int,
+        ascending: bool,
+        cursor: Optional[str] = None,
+    ) -> Connection[str]:
+        """Returns a connection object that contains a list of partition keys and all the necessary
+        information to paginate through them.
+
+        Args:
+            cursor: (Optional[str]): A cursor to track the progress paginating through the returned partition key results.
+            limit: (Optional[int]): The maximum number of partition keys to return.
+
+        Returns:
+            Connection[MultiPartitionKey]
+        """
+        current_time = context.temporal_context.effective_dt
+        dynamic_partitions_store = context.dynamic_partitions_store
+        multi_cursor = MultiPartitionCursor.from_cursor(cursor)
+        last_seen_key = multi_cursor.last_seen_key
+
+        dimension_keys = {}
+        for dimension in self._partitions_defs:
+            dimension_keys[dimension.name] = dimension.partitions_def.get_partition_keys(
+                current_time=current_time,
+                dynamic_partitions_store=dynamic_partitions_store,
+            )
+
+        if any(not results for results in dimension_keys.values()):
+            return Connection(results=[], cursor=multi_cursor.to_string(), has_more=False)
+
+        # Find the starting indices based on last_seen_key
+        dimension_start_indices = {}
+        for dim_name, results in dimension_keys.items():
+            if last_seen_key and dim_name in last_seen_key.keys_by_dimension:
+                try:
+                    idx = results.index(last_seen_key.keys_by_dimension[dim_name])
+                    dimension_start_indices[dim_name] = idx
+                except ValueError:
+                    # Key no longer exists, start from beginning for this dimension
+                    dimension_start_indices[dim_name] = 0
+            elif ascending:
+                dimension_start_indices[dim_name] = 0
+            else:
+                dimension_start_indices[dim_name] = len(results) - 1
+
+        dim_names = [dim.name for dim in self._partitions_defs]
+        current_indices = {
+            dim_name: dimension_start_indices.get(dim_name, 0) for dim_name in dim_names
+        }
+
+        def _invalid(indices):
+            return any(
+                indices[dim_name] >= len(dimension_keys[dim_name]) for dim_name in indices
+            ) or any(indices[dim_name] < 0 for dim_name in indices)
+
+        def _increment_indices():
+            for i in range(len(dim_names) - 1, -1, -1):
```

Comment:
Comment re: what `range(len(dim_names) - 1, -1, -1)` does would be useful. I had to use grok